src/Helper/HtmlAttributes.php

```diff
@@ -9,18 +9,14 @@
 
 /**
  * Helper for creating HtmlAttributesSet objects
- *
- * @final
  */
-class HtmlAttributes extends AbstractHelper
+final class HtmlAttributes
 {
-    use DeprecatedAbstractHelperHierarchyTrait;
-
-    private Escaper $escaper;
+    private readonly Escaper $escaper;
 
     public function __construct(?Escaper $escaper = null)
```

Comment:
BTW, since these helpers are always built via DIC, perhaps make the ctor argument mandatory, and remove instantiation?

It's just an idea / something I noticed

Reply:
Good point. If Escaper's constructor ever changes, however unlikely, this ensures deps are fully delegated 👍